src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,231 @@
+        return this.state === PLAYER_STATE.STOP;
+    }
+
+    isEnded() {
+        return this.state === PLAYER_STATE.END;
+    }
+
+    // Public functions
+
+    /**
+     * { Start animation }
+     *
+     * @param      {Animation} The animation to play
+     * @return     {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    play(animation) {
+        this.animation = animation;
+        if (this.animation.animate) this.callBackFrame = () => this.animation.animate(this.keyframe);
+        setPlayerState(this, PLAYER_STATE.PLAY);
+        resetTimer(this);
+        this.id = setInterval(this.frame.bind(this), INTERVAL_TIME);
+        this.promise = new Promise(r => this.resolve = r);
+        return this.promise;
+    }
+
+    /**
+     * { The animation is played after a number of frames }
+     *
+     * @param      {Animation}  animation    The animation to play
+     * @param      {Number}  waitingTime  The waiting time before start animation (time in frame)
+     */
+    playLater(animation, waitingFrame) {
+        this.resolveWait = null;
+        const promise = new Promise(r => this.resolveWait = r);
+        const timew = Math.floor(INTERVAL_TIME * waitingFrame);
+        window.clearInterval(this.waitTimer);
+        this.waitTimer = window.setTimeout(() => { this.play(animation).then(() => this.resolveWait()); }, timew);
+        return promise;
+    }
+
+    /**
+     * { Stop current animation }
+     *
+     * @return  {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    stop() {
+        setPlayerState(this, PLAYER_STATE.STOP);
+        finishAnimation(this);
+        this.dom.dispatchEvent(stopEvent);
+        // needed to return promise to wait sync
+        return Promise.resolve();
+    }
+
+    /**
+     * { Pause current animation }
+     */
+    pause() {
+        setPlayerState(this, PLAYER_STATE.PAUSE);
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    /**
+     * { this function is executed with each frame }
+     */
+    frame() {
+        if (this.keyframe < this.animation.duration) {
+            this.callBackFrame();
+            this.keyframe++;
+            this.dom.dispatchEvent(frameEvent);
+        }
+        else {
+            setPlayerState(this, PLAYER_STATE.END);
+            finishAnimation(this);
+            this.dom.dispatchEvent(endEvent);
+        }
+    }
+}
+
+/**
+ * { Animation }
+ * Animation is play by the AnimationPlayer during the time of duration
+ * During playback, the AnimationPlayer emits event for each frame
+ * Animation is used to execute a callback to each frame
+ * @class      Animation
+ * @param      {Number}  duration  The animation's duration in number of frames. FRAMERATE is number of frames in one seconde.
+ * @param      {String}  name      The animation's name. It's used for debug message.
+ */
+class Animation {
+    constructor(params) {
+        this.duration = params.duration || 0;
```

Comment:
Maybe pick a more sensible default value, 60 (1 sec) ? Or throw an error if `params.duration` is unset.